+nosnoc/+solver/MpccSolver.m

```diff
@@ -426,5 +426,443 @@ function print_infeasibility(obj, results)
                 keyboard
             end
         end
+
+        function [polished_w, res_out, stat_type, n_biactive] = calculate_stationarity(obj, results, exitfast, lifted)
+            import casadi.*
+            stat_type = "?";
+            nlp = obj.nlp;
+            mpcc = obj.mpcc;
+
+            % This method generates a TNLP (tightened NLP) using the results of a homotopy solver.
+            % In order to do so we first identify the active set of the complementarity constraints which involves identifying which
+            % branch of the complementarity each element of G(w) and H(w) are, or whether the solution is biactive at that point.
+            % For a more in depth explanation see the PhD theses of Armin Nurkanovic or Alexandra Schwartz
```

Comment:
We can be more specific with Pointing to Section 2.3 of https://publications.syscop.de/Nurkanovic2023f.pdf
and to https://arxiv.org/abs/2312.11022